// purplecheerio-wave/examples/taskserver.js
var url = require('url')

const util = require('util');

//app.get("/task0", function(req, res)
exports.gettask0 = function(req, res)
{
	var o = decodeURIComponent((url.parse(req.url).query));
	if(o)
		console.log("task0="+util.inspect(o));
	else
		console.log("task0");
	res.json({"f":"task0","r":0});
}
var debug_cErr  = 0
exports.gettask1 = function(req, res)
{
	debug_cErr++;
	// To test retry functionality increase this to more than 1
	if(debug_cErr < 1)
	{
		console.log("error condition");
		res.status(500).send("forced retry");
		return;
	}
	var o = decodeURIComponent((url.parse(req.url).query));
	if(o)
		console.log("task1="+util.inspect(o));
	else
		console.log("task1");
	res.json({"f":"task1","r":0});
}
exports.gettask2 = function(req, res)
{
	var o = decodeURIComponent((url.parse(req.url).query));
	if(o)
		console.log("task2="+util.inspect(o));
	else
		console.log("task2");
	res.json({"f":"task2","r":0});
}
exports.gettask0000000000 = function(req, res)
{
	var o = decodeURIComponent((url.parse(req.url).query));
	if(o)
		console.log("task0000000000="+util.inspect(o));
	else
		console.log("task0000000000");

	res.json({"f":"task0000000000","r":1});
}
exports.posttask3 = function(req, res)
{
	console.log("task3="+util.inspect(req.body));
	res.json({"f":"task3","r":0});
}
exports.gettask4 = function(req, res)
{
	var o = decodeURIComponent((url.parse(req.url).query));
	if(o)
		console.log("task4="+util.inspect(o));
	else
		console.log("task4");
	res.json({"f":"task4","r":0});
}

exports.gettask5 = function(req, res)
{
	var o = decodeURIComponent((url.parse(req.url).query));
	if(o)
		console.log("task5="+util.inspect(o));
	else
		console.log("task5");
	res.json({"f":"task5","r":0});
}

exports.gettask6 = function(req, res)
{
	var o = decodeURIComponent((url.parse(req.url).query));
	if(o)
		console.log("task6="+util.inspect(o));
	else
		console.log("task6");
	res.json({"f":"task6","r":0});
}
